### PR TITLE
[SPARK-36181][PYTHON] Update pyspark sql readwriter documentation

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
Updating the pyspark sql readwriter documentation to the level of detail provided by the scala documentation

### Why are the changes needed?
documentation clarity

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Only documentation change

### CODE REVIEW

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -624,7 +624,8 @@ def partitionBy(self, *cols):
 
     def bucketBy(self, numBuckets, col, *cols):
         """Buckets the output by the given columns.If specified,
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:626]:
```suggestion
        """Buckets the output by the given columns. If specified,
```

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -742,10 +743,16 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
     def insertInto(self, tableName, overwrite=None):
         """Inserts the content of the :class:`DataFrame` to the specified table.
 
+        Optionally overwriting any existing data.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:746]: Can you convert this to `Parameters`? You can just write down the description like, this is equivalent with `DataFrameWrtier.mode("overwrite")`

File: python/pyspark/sql/readwriter.py [hunk 3]
```diff
@@ -742,10 +743,16 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
...
+
         It requires that the schema of the :class:`DataFrame` is the same as the
         schema of the table.
 
-        Optionally overwriting any existing data.
+        Notes
+        -----
+        Unlike `saveAsTable`, `insertInto` ignores the column names and just uses
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:753]: Let's switch these to `` :meth:`DataFrameWriter.saveAsTable` ``, etc.

File: python/pyspark/sql/readwriter.py [hunk 4]
```diff
@@ -766,6 +773,13 @@ def saveAsTable(self, name, format=None, mode=None, partitionBy=None, **options)
 
         .. versionadded:: 1.4.0
 
+        Notes
+        -----
+        When `mode` is `Append`, if there is an existing table, we will use the format and
+        options of the existing table. The column order in the schema of the `DataFrame` doesn't
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:779]:
```suggestion
        options of the existing table. The column order in the schema of the :class:`DataFrame` doesn't
```

File: python/pyspark/sql/readwriter.py [hunk 5]
```diff
@@ -766,6 +773,13 @@ def saveAsTable(self, name, format=None, mode=None, partitionBy=None, **options)
...
+        need to be same as that of the existing table. Unlike `insertInto`, `saveAsTable` will use
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:780]: ditto for insertInto

**Comment by HyukjinKwon** [comment]
Looks good otherwise

File: python/pyspark/sql/readwriter.py (hunk 2 above)
Comment (dominikgehl) [reply python/pyspark/sql/readwriter.py]: Done

File: python/pyspark/sql/readwriter.py (hunk 4 above)
Comment (dominikgehl) [reply python/pyspark/sql/readwriter.py]: Done

File: python/pyspark/sql/readwriter.py (hunk 3 above)
Comment (dominikgehl) [reply python/pyspark/sql/readwriter.py]: Done

File: python/pyspark/sql/readwriter.py (hunk 5 above)
Comment (dominikgehl) [reply python/pyspark/sql/readwriter.py]: Done

File: python/pyspark/sql/readwriter.py (hunk 1 above)
Comment (dominikgehl) [reply python/pyspark/sql/readwriter.py]: Done

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2.

**Pull request closed without merging**